janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/CustomAnalyzer.java

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.diskstorage.lucene;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.DelegatingAnalyzerWrapper;
+import org.janusgraph.core.schema.Mapping;
+import org.janusgraph.core.schema.Parameter;
+import org.janusgraph.diskstorage.indexing.KeyInformation;
+import org.janusgraph.graphdb.types.ParameterType;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Olivier binda (olivier.binda@wanadoo.fr)
+ */
+
+/**
+ * JanusGraph may open different index stores and needs an analyzer for each
+ * so an analyzer should be provided to build IndexWriters/IndexReaders that are used when an index is opened and documents are added
+ * BUT a janusGraph may add a new indexable field, with a custom tokenizing process through Parameters, at any moment, which means that it should be possible
+ * to map any field to the corresponding custom analyzer, though indexreaders are already in use.
+ * <p>
+ * A way to achieve that would be to use a delegating analyzer that delegates Tokenizing for a field to
+ * the corresponding Analyzer (whose class you get through a Mapping)
+ */
+public class CustomAnalyzer extends DelegatingAnalyzerWrapper {
+    private static String KEYWORD_ANALYZER = org.apache.lucene.analysis.core.KeywordAnalyzer.class.getName();
```

Comment:
final

Reply:
Done. I'll also rebase these into a single commit too.

Many thanks for the kind review. 
Best regards,
Olivier